### ISSUE TITLE
What does "traversal" mean in Step 2.B?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/accname#79]
> 2.B: if computing a name, and the current node has an aria-labelledby attribute that contains at least one valid IDREF, and the current node is not already part of an aria-labelledby traversal

Does this refer to traversal in general (aria-labelledby cannot reference an element in which it is located) or only to traversal in relation to aria-labelledby (aria-labelledby is ignored if the element is already referenced by aria-labelledby)?

Look at: 
```
<button id="test">
  1
  <span aria-labelledby="test alb">span</span>
  2
</button>
<span id="alb">alb</span>
```

The name of the button should be according to accname: 1 1 span 2 alb 2

Explanation: we have 3 child nodes inside the button:
* text node: 1
* span node: labelled with the buttons name and the span outside: 
	* 1 (text node)
	* span (no further traversal, thus text node)
	* 2 (text node)
	* alb (aria-labelledby)
* text node: 2

However, all tested browsers and https://whatsock.github.io/w3c-alternative-text-computation/Editable%20Live%20Input%20AccName%20Test.html come to a different result: 

* https://whatsock.github.io/w3c-alternative-text-computation/Editable%20Live%20Input%20AccName%20Test.html: 1 1 span 2 alb
* Firefox: 1 span 2
* IE: 1 span 2
* Chrome: 1 1 2 alb 2

**Comment by accdc** [comment]
Hi,
In this case, you are coming across different rules that are applied, and they are still handled differently between browsers, mainly because it's complicated and subject to bugs more than anything.

First, traversal simply means that you can only follow an aria-labelledby reference once for a specific id, otherwise your example would cause an infinite loop and crash the browser by repeatedly processing id=test over and over again forever.

Second, you can only process a node once during the recursion process, so if that node has already been processed, then that node should be skipped over if it is processed again as part of the same root node process. Since you have only one instance of "1" there, then the WhatSock one is incorrect and should only say that once as "1 span 2 alb" for the full name, because 1 was already parsed, and should not be parsed again the second time through.

I will look into this tomorrow and see why it's not doing that.

**Comment by JAWS-test** [comment]
@accdc 

Thank you very much for your answer. However, I see two problems in this context:

* this is not in the specification and should be added
* your remarks do not correspond with [example 1](https://w3c.github.io/accname/#example-1). According to what you said, example 1 wouldn't be traversal. So the example should be changed
